src/analysis/analytic/isolated_zeros.lean

```diff
@@ -137,4 +140,41 @@ begin
   { exact or.inr (hp.locally_ne_zero h) }
 end
 
+lemma frequently_zero_iff_eventually_zero {f : 𝕜 → E} {w : 𝕜} (hf : analytic_at 𝕜 f w) :
+  (∃ᶠ z in 𝓝[≠] w, f z = 0) ↔ (∀ᶠ z in 𝓝 w, f z = 0) :=
+⟨hf.eventually_eq_zero_or_eventually_ne_zero.resolve_right,
+  λ h, (h.filter_mono nhds_within_le_nhds).frequently⟩
+
 end analytic_at
+
+namespace analytic_on
+
+variables {U : set 𝕜} {w : 𝕜}
+
+theorem eq_on_of_preconnected_of_frequently_eq (hf : analytic_on 𝕜 f U)
+  (hU : is_preconnected U) (hw : w ∈ U) (hfw : ∃ᶠ z in 𝓝[≠] w, f z = 0) :
```

Comment:
Can you add a version with `hfw : w ∈ closure ({z | f z = 0} \ {w})`? I think this is equivalent to the original `hfw` and both say that `w` is a limit/accumulation/cluster point of {z | f z = 0}. You could probably convert between them using [mem_closure_iff_frequently](https://leanprover-community.github.io/mathlib_docs/topology/basic.html#mem_closure_iff_frequently) + a little more. `cluster_pt` in mathlib is defined for an arbitrary (not necessarily principal) filter, and even when applied to a principal filter, it doesn't agree with the standard notion of cluster point unfortuantely (see [mem_closure_iff_cluster_pt](https://leanprover-community.github.io/mathlib_docs/topology/basic.html#mem_closure_iff_cluster_pt), which involves the closure of the whole set instead of the complement of the point). Maybe we should develop some theory about the standard cluster points?

Reply:
I can add such a version or at least the lemma showing that both `hfw` versions are indeed equivalent. But I believe that in terms of API it would be more useful to have a version asserting `f z = g z` along a set with an accumulation point, concluding `eq_on U f g`, as there will be a little more plumbing involved and it serves often in practice.

Reply:
Added!